src/Microsoft.AspNet.Authentication.JwtBearer/JwtBearerHandler.cs

```diff
@@ -150,23 +159,25 @@ internal class JwtBearerHandler : AuthenticationHandler<JwtBearerOptions>
                 await Options.Events.AuthenticationFailed(authenticationFailedContext);
                 if (authenticationFailedContext.HandledResponse)
                 {
-                    return authenticationFailedContext.AuthenticationTicket;
+                    return new AuthenticateResult()
+                    {
+                        Ticket = authenticationFailedContext.AuthenticationTicket
+                    };
                 }
 
                 if (authenticationFailedContext.Skipped)
                 {
-                    return null;
+                    return new AuthenticateResult();
                 }
 
                 throw;
```

Comment:
I'm not opposed to re-throwing exceptions by default. But they MUST be caught somewhere: we can't afford returning a 500 response for a simple invalid/expired token.